### ISSUE TITLE
[Android] getCurrentPosition with coarse location permission only

### ISSUE THREAD

**Issue by emorb** [transistorsoft/react-native-background-geolocation#2041]
Hi,
I noticed some weird behavior in Android when using getCurrentPosition() method with a timeout and providing coarse location permission only.

## Your Environment
* Plugin version: 4.16.2
* Platform: Android
* OS version: 12
* Device manufacturer / model: Google Pixel 3a
* React Native version (`react-native -v`): 0.72.4
* Plugin config
```typescript <!-- Syntax-highlighting:  paste your code below -->
BackgroundGeolocation.getCurrentPosition({
  timeout: 5,
  samples: 3,
  desiredAccuracy: 50 // even using 100000 the result is the same
})
  .then((position) => {
    console.log(JSON.stringify(position, null, 2))
  })
  .catch((e) => {
    console.error(e)
  })
```

## Expected Behavior
I expect getCurrentPosition() to ask user coarse or fine location permissions the first time it is invoked.
Since providing coarse position permission is enough for the first call, I expect it to be enough even for the further ones, without asking user to provide fine location (maybe such a behavior should be controlled by a flag or something in the request params).

## Actual Behavior
When invoking getCurrentPosition() for the first time, if user only grants coarse location permission, location is returned. Invoking getCurrentPosition() again, a native alert appears asking user to provide fine location.
Moreover, if a timeout is provided, the popup is displayed but a location is returned before user actually chooses to provide fine one or not, making the fine permission request irrelevant. 

## Steps to Reproduce

1. Invoke getCurrentPosition()
2. Grant coarse location permission only
3. Invoke getCurrentLocation() again
4. Just wait without closing the native dialog

## Context

I want to ask user location accuracy only once, if at least coarse location permission is provided

## Debug logs

<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->
PASTE_YOUR_LOGS_HERE
```

</details>

**Comment by christocracy** [comment]
.getCurrentPosition always uses highest possible accuracy.

the options.desiredAccuracy has a different meaning with .getCurrentPosition — it means the accuracy you’ll be satisfied with when providing options.samples, where if a location having accuracy <= options.desiredAccuracy arrives, the method will stop further sampling and deliver that location.

**Comment by emorb** [comment]
Thanks for your answer and sorry to resume this thread again, I just need a confirmation:

On Android, when invoking getCurrentPosition method:
1. If FINE_LOCATION_PERMISSION is granted, I will receive a position
2. If COARSE_LOCATION_PERMISSION only is granted, I will receive an Error 1 - Permission Denied

On iOS:
1. If permission is granted with Precise: ON, I will receive a position
2. If permission is granted with Precise: OFF, I will receive a position

Is it possible configure plugin params or request params to retrieve a position (ofc not a precise one) in Android also if only COARSE_LOCATION_PERMISSION is granted?

Thanks

**Comment by christocracy** [comment]
Try configuring locationAuthorizationRequest: “Any”

**Comment by emorb** [comment]
I tried this configuration when initializing the plugin:
```typescript
export const defaultConfig: Config = {
  logLevel: BackgroundGeolocation.LOG_LEVEL_OFF,
  debug: false,
  startOnBoot: false,
  disableMotionActivityUpdates: true,
  desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_MEDIUM,
  locationAuthorizationRequest: 'Any',
}
```

and these params to .getCurrentPosition():
```typescript
{
  extras: { getCurrentPosition: true },
  maximumAge: undefined,
  persist: false,
  samples: 3,
  timeout: 10,
}
```
but I'm still getting Error 1.

I also tried to add a callback .onLocation() and what I get is a { error: "408" } response

I also updated to React-native 0.73.8 and the latest version of plugin 4.17.2

**Comment by christocracy** [comment]
Do you know what error code 408 means?

are you observing the plug-in logs in real time?  See wiki “Debugging”,

**Comment by emorb** [comment]
Yes. I'm observing logs in real time. I checked Debugging section and also improved logs.
But everything seems fine to me, I also removed the timeout params (so it should default to 30secs) but, even doing this, I get same error and timeout seems to be triggered too early:

1. Initializing GPS with 
```
{
  "debug": false, 
  "desiredAccuracy": 100, 
  "disableMotionActivityUpdates": true, 
  "locationAuthorizationRequest": "Any", 
  "logLevel": 0, "startOnBoot": false}
```
2. requesting GPS position at **2024-10-16T15:07:56.408Z** with 
```
{
  "extras": {
    "getCurrentPosition": true
  }, 
  "maximumAge": undefined, 
  "persist": false, 
  "samples": 3
} 
```
3. getCurrentPosition error 1 at **2024-10-16T15:07:58.692Z**
4. on location callback returned (in success callback, not in failure one) 408 at **2024-10-16T15:08:57.709Z**
